pkg/logs/sds/scanner.go

```diff
@@ -307,20 +309,27 @@ func interpretRCRule(userRule RuleConfig, standardRule StandardRuleConfig, defau
 	}
 
 	if defToUse.Version == -1 {
-		// TODO(remy): telemetry
 		return nil, fmt.Errorf("unsupported rule with no compatible definition")
 	}
 
-	// we use the filled `CharacterCount` value to decide if we want
-	// to use the user provided configuration for proximity keywords
-	// or if we have to use the information provided in the std rules instead.
-	if userRule.IncludedKeywords.CharacterCount > 0 {
-		// proximity keywords configuration provided by the user
-		extraConfig.ProximityKeywords = sds.CreateProximityKeywordsConfig(userRule.IncludedKeywords.CharacterCount, userRule.IncludedKeywords.Keywords, nil)
-	} else if len(defToUse.DefaultIncludedKeywords) > 0 && defaults.IncludedKeywordsCharCount > 0 {
-		// the user has not specified proximity keywords
-		// use the proximity keywords provided by the standard rule if any
+	// If the "Use recommended keywords" checkbox has been checked, we use the default
+	// included keywords available in the rule (curated by Datadog).
+	// Otherwise:
+	//   If some included keywords have been manually filled by the user, we use them
+	//   Else we start using the default excluded keywords.
+	if userRule.IncludedKeywords.UseRecommendedKeywords {
+		// default included keywords
 		extraConfig.ProximityKeywords = sds.CreateProximityKeywordsConfig(defaults.IncludedKeywordsCharCount, defToUse.DefaultIncludedKeywords, nil)
+	} else {
+		if len(userRule.IncludedKeywords.Keywords) > 0 && userRule.IncludedKeywords.CharacterCount > 0 {
+			// user provided included keywords
+			extraConfig.ProximityKeywords = sds.CreateProximityKeywordsConfig(userRule.IncludedKeywords.CharacterCount, userRule.IncludedKeywords.Keywords, nil)
+		} else if len(defaults.ExcludedKeywords) > 0 && defaults.ExcludedKeywordsCharCount > 0 {
+			// default excluded keywords
+			extraConfig.ProximityKeywords = sds.CreateProximityKeywordsConfig(defaults.ExcludedKeywordsCharCount, nil, defaults.ExcludedKeywords)
```

Comment:
I think to mimic the behavior applied in SDS cloud, the logic should be a bit different.

If there is no default keywords (`len(defaults.IncludedKeywordsCharCount) == 0`), then SDS should rely on default excluded keywords.

In other words:
```
if (`len(defaults.IncludedKeywordsCharCount) == 0`) and `len(userRule.IncludedKeywords.Keywords) == 0`) 
then use excluded keywords. 
else 
 rely on included keywords config (same as the current logic)
```